### PR TITLE
Build error when add sms auto fill fix, android x dependecy added, small refactoring

### PR DESCRIPTION
**Pull request by BeDaut**
Build error when add sms auto fill fix, android x dependecy added, small refactoring

### CODE REVIEW

File: android/src/main/java/com/jaumard/smsautofill/AppSignatureHelper.java [hunk 1]
```diff
@@ -73,7 +74,7 @@ private static String hash(String packageName, String signature) {
         String appInfo = packageName + " " + signature;
         try {
             MessageDigest messageDigest = MessageDigest.getInstance(HASH_TYPE);
-            messageDigest.update(appInfo.getBytes(StandardCharsets.UTF_8));
+            messageDigest.update(appInfo.getBytes(Charset.forName("UTF-8")));
```
Comment (jaumard) [inline android/src/main/java/com/jaumard/smsautofill/AppSignatureHelper.java]: why this change ? I rather not depend on a string if a constant exist

Comment (BeDaut) [reply android/src/main/java/com/jaumard/smsautofill/AppSignatureHelper.java]: `StandardCharsets.UTF_8` is not compatible with older android versions

Comment (jaumard) [reply android/src/main/java/com/jaumard/smsautofill/AppSignatureHelper.java]: Ho ok :) LGTM then ;) Thanks a lot !
